app/src/main/java/com/rage/clamber/Fragments/CommentsFragment.java

```diff
@@ -0,0 +1,187 @@
+                             Bundle savedInstanceState) {
+
+        View rootView = inflater.inflate(R.layout.fragment_comments, container, false);
+        ButterKnife.bind(this, rootView);
+
+        mainUser = getArguments().getParcelable(HomePage.ARG_USER);
+        climbId = getArguments().getInt(ClimbsRecyclerViewAdapter.ARG_CLIMB);
+        comments = new ArrayList<>();
+        recyclerView.setLayoutManager(new LinearLayoutManager(getActivity()));
+        adapter = new CommentsRecyclerViewAdapter(comments, mainUser, getContext());
+        recyclerView.setAdapter(adapter);
+
+        getCommentsByClimb(climbId);
+
+        return rootView;
+    }
+
+    /**
+     * Method to make a network call to the Clamber Server to get all Comments by Climb Id. The
+     * climbsRecyclerViewAdapter has an OnClick method on the comments button that launches this
+     * fragment and triggers this network call to display comments.
+     *
+     * @param climbId - id of the climb that was selected.
+     */
+    public void getCommentsByClimb(int climbId) {
+        ConnectivityManager connectivityManager = (ConnectivityManager) getActivity().getSystemService(Context.CONNECTIVITY_SERVICE);
+        NetworkInfo networkInfo = connectivityManager.getActiveNetworkInfo();
+        if (networkInfo != null && networkInfo.isConnected()) {
+            Call<List<Comment>> commentsCall = ApiManager.getClamberService().getComments(climbId);
+            commentsCall.enqueue(new Callback<List<Comment>>() {
+                @Override
+                public void onResponse(Call<List<Comment>> call, Response<List<Comment>> response) {
+                    if (response.code() == 200) {
+                        List<Comment> commentsList = response.body();
+                        comments.addAll(commentsList);
+                        if (comments.isEmpty()) {
+                            noCommentsText.setVisibility(View.VISIBLE);
+                        }
+
+                    } else {
+                        Log.d(TAG, "Non 200 response code returned - check server");
+                    }
+
+                    adapter.notifyDataSetChanged();
+                }
+
+                @Override
+                public void onFailure(Call<List<Comment>> call, Throwable t) {
+                    Log.d(TAG, "Failure when attempting to return comments for climb", t);
+
+                }
+            });
+
+        } else {
+            Toast.makeText(getContext(), R.string.no_internet_connection, Toast.LENGTH_SHORT).show();
+        }
+    }
+
+    /**
+     * OnClick for the "add comment" button. Takes the text in the EditText field and Posts it to
+     * the database. Prints out a log message if it is unsuccessful.
+     */
+    @OnClick(R.id.comments_fragment_add_comment_button)
+    public void onAddCommentButtonClicked() {
+
+        AddCommentDialogFragment dialogFragment = new AddCommentDialogFragment();
+        dialogFragment.setTargetFragment(this, 0);
```

Comment:
probably worth leaving a little comment here explaining why you need `setTargetFragment` here